scipy/special/_basic.py

```diff
@@ -3087,6 +3090,138 @@ def factorialk(n, k, exact=True):
     return _exact_factorialx_array(n, k=k)
 
 
+def stirling2(N, K, *, exact=True):
+    """Generate Stirling number(s) of the second kind.
+
+        Stirling numbers of the second kind count the number of ways to
+        partition a set with N elements into K non-empty subsets.
+
+        The values this function returns are calculated using a dynamic
+        program which avoids redundant computation across the subproblems
+        in the solution. For array-like input, this implementation also 
+        avoids redundant computation across the different Stirling number
+        calculations.
+
+        The numbers are sometimes denoted
+        .. math::
+
+            `{N \brace K}`;
+
+        see [GKP]_ for details. This is often expressed-verbally-as
```

Comment:
```suggestion
        see [1]_ for details. This is often expressed-verbally-as
```

Reply:
fwiw, we do have author initials used for references in other places in special, https://github.com/scipy/scipy/blob/876c68da9dfa6d91f110c7f85d621a9177b89eb0/scipy/special/_add_newdocs.py#L3627-L3633, but it is probably good to make this conform to the most common case